lib/Sapi.php

```diff
@@ -67,7 +67,21 @@ static function sendResponse(ResponseInterface $response) {
             }
 
         }
-        file_put_contents('php://output', $response->getBody());
+
+        $body = $response->getBody();
+        if(is_null($body)) return;
+
+        $contentLength = $response->getHeader('Content-Length');
+        if($contentLength !== null) {
+            $output = fopen('php://output', 'wb');
+            if(is_resource($body) && get_resource_type($body) == 'stream') {
+                stream_copy_to_stream($body, $output, $contentLength);
```

Comment:
random side-note: do we need special handling for compressed requests, because those use Content-Length differently IIRC?